vivarium/simulator/states.py

```diff
@@ -0,0 +1,316 @@
+        return self.nve_state.entity_type == etype.value
+
+    def row_idx(self, field, nve_idx):
+        return nve_idx if field == 'nve_state' else self.nve_state.entity_idx[jnp.array(nve_idx)]
+
+    def __getattr__(self, name):
+        def wrapper(e_type):
+            value = getattr(self.nve_state, name)
+            if isinstance(value, rigid_body.RigidBody):
+                return rigid_body.RigidBody(center=value.center[self.e_cond(e_type)],
+                                            orientation=value.orientation[self.e_cond(e_type)])
+            else:
+                return value[self.e_cond(e_type)]
+        return wrapper
+
+
+# Helper function to transform a color string into rgb with matplotlib colors
+def _string_to_rgb(color_str):
+    return jnp.array(list(mcolors.to_rgb(color_str)))
+
+
+def init_simulator_state(
+        box_size: float = 100.,
+        n_agents: int = 10,
+        n_objects: int = 2,
+        num_steps_lax: int = 4,
+        dt: float = 0.1,
+        freq: float = 40.,
+        neighbor_radius: float = 100.,
+        to_jit: bool = True,
+        use_fori_loop: bool = False
+        ) -> SimulatorState:
+    """
+    Initialize simulator state with given parameters
+    """
+    return SimulatorState(
+        idx=jnp.array([0]),
+        box_size=jnp.array([box_size]),                       
+        n_agents=jnp.array([n_agents]),
+        n_objects=jnp.array([n_objects]),
+        num_steps_lax=jnp.array([num_steps_lax], dtype=int),
+        dt=jnp.array([dt], dtype=float),
+        freq=jnp.array([freq], dtype=float),
+        neighbor_radius=jnp.array([neighbor_radius], dtype=float),
+        # Use 1*bool to transform True to 1 and False to 0
+        to_jit= jnp.array([1*to_jit]),
+        use_fori_loop=jnp.array([1*use_fori_loop]))
+
+
+def _init_positions(key_pos, positions, n_elements, box_size, n_dims=2):
+    assert (len(positions) == n_elements if positions else True)
+    # If positions are passed, transform them in jax array
+    if positions:
+        positions = jnp.array(positions)
+    # Else initialize random positions
+    else:
+        positions = random.uniform(key_pos, (n_elements, n_dims)) * box_size
+    return positions
+
+def _init_existing(n_existing, n_elements):
+    if n_existing:
+        assert n_existing <= n_elements
+        existing_arr = jnp.ones((n_existing))
+        non_existing_arr = jnp.zeros((n_elements - n_existing))
+        exists_array = jnp.concatenate((existing_arr, non_existing_arr))
+    else:
+        exists_array = jnp.ones((n_elements))
+    return exists_array
+
+# TODO : Should also add union float, list[float] for friction, diameter ..
+# Also think it would be easier to only handle the nve state in the physics engine. (if it doesn't make the  simulation run slower)
+# Here it makes a really long function, and it isn't very modular 
+# Plus we store different attributes of entities (e.g Agents positions and sensors) in different dataclasses
```

Comment:
Let's discuss it, as we said last time it is part of a more general issue about how we handle conversions between data types that encode the simular infomation (AgentState, NVEState, Config, proto ..)

Reply:
You can create an issue about explaining what we have already discussed on this topic (a potential solution would be to have useful helper functions in the State class)